src/redux/books/books.js

```diff
@@ -0,0 +1,19 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const books = createSlice({
+  name: 'book',
+  initialState: { books: [] },
+  reducers: {
+    addBook(state, action) {
+      state.push(action.payload.book);
+    },
+    removeBook(state, action) {
+      // eslint-disable-next-line no-param-reassign
+      state.books = state.books.filter((book) => book.id !== action.payload.id);
+    },
+  },
+});
+
+export const booksActions = books.actions;
+
+export default books;
```

Comment:
- [ ] I can see that you used `createSlice` from `toolkit` to write your book's actions and reducers and this is not wrong, But according to the requirements, you should write your book's actions and reducer using the [ducks pattern](https://github.com/erikras/ducks-modular-redux) and the value of each action type needs to follow the ducks pattern. So kindly recreate this file using [ducks pattern](https://github.com/erikras/ducks-modular-redux) to follow the requirements 💯👌

> It should be something like

```
// Define Action Type
const ADD = 'bookstore/books/ADD';
const REMOVE = 'bookstore/books/REMOVE';

export default function booksReducer(state = [], action) {
  switch (action.type) {
    case ADD: return ... ;
    case REMOVE: return ... ;
    default: return state;
  }
}

export function addBook(book) { ... }

export function removeBook(id) { ... }
```

![-2](https://user-images.githubusercontent.com/45231276/184617059-a7187888-29c6-4ba1-8d71-447bc33cd965.PNG)